supportedBackends/sfs/src/main/scala/cromwell/backend/impl/sfs/config/ConfigHashingStrategy.scala

```diff
@@ -87,10 +91,61 @@ final case class HashPathModTimeStrategy(checkSiblingMd5: Boolean) extends Confi
   override val description = "hash file path and last modified time"
 }
 
-final case class HashFileStrategy(checkSiblingMd5: Boolean) extends ConfigHashingStrategy {
+final case class HashFileMd5Strategy(checkSiblingMd5: Boolean) extends ConfigHashingStrategy {
   override protected def hash(file: Path): Try[String] = {
     tryWithResource(() => file.newInputStream) { DigestUtils.md5Hex }
   }
 
-  override val description = "hash file content"
+  override val description = "hash file content with md5"
 }
+
+final case class HashFileXxH64Strategy(checkSiblingMd5: Boolean) extends ConfigHashingStrategy {
+  override protected def hash(file: Path): Try[String] = {
+    tryWithResource(() => file.newInputStream) {HashFileXxH64StrategyMethods.xxh64sum(_)}
+  }
+  override val description = "hash file content with xxh64"
+}
+
+final case class FingerprintStrategy(checkSiblingMd5: Boolean) extends ConfigHashingStrategy {
+  override protected def hash(file: Path): Try[String] = {
+    Try {
+      file.lastModifiedTime.toEpochMilli.toHexString +
+      file.size.toHexString +
+      // Only check first 10 MB (10485760 bytes) for performance reasons
+      HashFileXxH64StrategyMethods.xxh64sum(file.newInputStream, maxSize = 10485760L)
+      }
+    }
+  override val description = "fingerprint the file with last modified time, size and a xxh64 hash of the first 10 mb"
+}
+
+object HashFileXxH64StrategyMethods {
+  // For more information about the choice of buffer size: https://github.com/rhpvorderman/hashtest/
+  private lazy val defaultBufferSize: Int = 128 * 1024
+  private lazy val xxhashFactory: XXHashFactory = XXHashFactory.fastestInstance()
+
+  /**
+    * Returns the xxh64sum of an input stream. The input stream is read in a buffered way.
+    * @param inputStream an input Stream
+    * @param bufferSize the size in bytes for the buffer.
+    * @param maxSize, only calculate the hash for the first maxSize bytes. Must be a multiply of bufferSize.
+    * @return A hex string of the digest.
+    */
+  def xxh64sum(inputStream: InputStream,
+               bufferSize: Int = defaultBufferSize,
+               maxSize: Long = Long.MaxValue): String = {
+    val hasher = xxhashFactory.newStreamingHash64(0)
+    val buffer: Array[Byte] = new Array[Byte](bufferSize)
+    var byteCounter: Long = 0
+    try {
```

Comment:
No `catch` here makes me nervous...

Incidentally/additionally, you might find the `cromwell.util.TryWithResource` method useful.

Reply:
Added a catch statement here that catches all throwables and exceptions. Weirdly, Throwable does not inherit from Exception :confused: . So I just catch both now to be sure.

Reply:
Indeed I think the default behavior for `try ... finally` is to raise errors, but execute finally nevertheless. But I was not so sure about this crashing thing. It should raise errors when occurring obviously. I have removed `catch`.